File: framework/src/main/java/com/bytex/snamp/connector/health/OkStatus.java
package com.bytex.snamp.connector.health;

import javax.annotation.Nonnull;
import java.time.Instant;
import java.util.Locale;

/**
 * Indicates that everything is OK.
 * @author Roman Sakno
 * @version 2.1
 * @since 2.0
 */
public final class OkStatus extends HealthStatus {
    private static final long serialVersionUID = 5391122005596632004L;

    public OkStatus(final Instant timeStamp) {
        super(timeStamp);
    }

    @Override
    public boolean like(final HealthStatus status) {
        return status instanceof OkStatus;
    }

    public OkStatus(){
        this(Instant.now());
    }

    public static boolean notOk(final HealthStatus status){
        return !(status instanceof OkStatus);
    }

    /**
     * Selects the worst health status between two statuses.
     *
     * @param other Other health status.
     * @return The worst health status.
     */
    @Override
    public HealthStatus worst(@Nonnull final HealthStatus other) {
        return other;
    }

    @Override
    public String toString(final Locale locale) {
        return "Everything is fine";
    }
}

File: webconsole/src/main/resources/_webapp/src/app/services/model/healthstatus/ok.status.ts
import { HealthStatus } from './health.status';

export class OkStatus extends HealthStatus {

    getNotificationLevel(): string {
        return "ok";
    }

    public represent():string {
        return "Everything is fine";
    }

    public getShortDescription():string {
        return "n/a";
    }

    public htmlDetails():string {
        let _details:string =  "<strong>Everything is fine</strong></br>";
        if (this.serverTimestamp.length > 0) {
            _details += "<strong>Server timestamp: </strong>" + this.serverTimestamp + "<br/>";
        }
        return _details;
    }
}